feat: DIA-1748: 'Generate labels with AI' button in Project, in SC

niklub opened the pull request:
### PR fulfills these requirements
- [ ] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [ ] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend



### Describe the reason for change
_(link to issue, supportive screenshots etc.)_



#### What does this fix?
_(if this is a bug fix)_



#### What is the new behavior?
_(if this is a breaking or feature change)_



#### What is the current behavior?
_(if this is a breaking or feature change)_



#### What libraries were added/updated?
_(list all with version changes)_



#### Does this change affect performance?
_(if so describe the impacts positive or negative)_



#### Does this change affect security?
_(if so describe the impacts positive or negative)_



#### What alternative approaches were there?
_(briefly list any if applicable)_



#### What feature flags were used to cover this change?
_(briefly list any if applicable)_



### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_


- https://github.com/HumanSignal/label-studio-sdk/pull/372 linked by [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/12431264450)

Commit:
feat: DIA-1748: 'Generate labels with AI' button in Project, in SC

Commit:
Add signals.py

Commit:
Add infobanner

Commit:
Revert data import

Commit:
Revert toolbar

Commit:
Fix contextlog

Commit:
Fix order of project.save()

Commit:
Merge remote-tracking branch 'origin/develop' into fb-dia-1748

Commit:
Add UserTour model

Commit:
Update states

Commit:
Merge remote-tracking branch 'origin/develop' into fb-dia-1748

Commit:
update data model, add url

robot-ci-heartex commented:
Follow Merge downstream workflow has been failed.
> [Workflow run](https://github.com/HumanSignal/label-studio-enterprise/actions/runs/12382379751)

Commit:
Merge remote-tracking branch 'origin/develop' into fb-dia-1748

Commit:
Optimize /api/templates

Commit:
Fix tests

Commit:
Fix url

Commit:
Fix tests

Commit:
Remove infobanner

Commit:
Update content

Commit:
Fix serialiser 'name' error

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12431264450

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12432646030

matt-bernstein committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12434128448

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12435080633

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12436508850

triklozoid commented on label_studio/users/product_tours/models.py:
```diff
@@ -0,0 +1,53 @@
+from pydantic import BaseModel, Field
+
+
+class ProductTourState(str, Enum):
```
We use `TextChoices` for that: https://github.com/HumanSignal/label-studio-enterprise/blob/develop/label_studio_enterprise/assignments/models.py#L15

niklub replied:
fixed

triklozoid commented on label_studio/projects/models.py:
```diff
@@ -740,11 +750,15 @@ def save(self, *args, update_fields=None, recalc=True, **kwargs):
+            self.__original_label_config = self.label_config
+            # if tasks are already imported, emit signal that project is configured and ready for labeling
+            if self.num_tasks > 0:
+                ProjectSignals.post_label_config_and_import_tasks.send(sender=Project, project=self)
```
This call can affect performance and stability of project saving, I would call it async or at least add `try:except:`

niklub replied:
the performance overhead for handling this particular call is subtle. Although it makes sense to avoid any pitfalls in the corresponding handlers, I'll put checks there

triklozoid commented on label_studio/users/product_tours/api.py:
```diff
@@ -0,0 +1,44 @@
+            tour_serializer.is_valid(raise_exception=True)
+            tour = tour_serializer.save()
+        else:
+            logger.info(f'Product tour {name} requested for user {self.request.user.id}.')
```
`info`? It will be logged on every call.

niklub replied:
changed to debug

Commit:
Address comments

Commit:
Update joyride content

Commit:
Ignore migrations in linter

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12496970745

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12497084177

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12497120230

Commit:
Fix target selector

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12525672032

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12549414497

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12596963299

Commit:
Merge remote-tracking branch 'origin/develop' into fb-dia-1748

Commit:
Update tour content

niklub commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12629966388)
> Successfully merged:  4 files changed, 40 insertions(+), 20 deletions(-)

niklub committed:
Merge branch 'develop' into 'fb-dia-1748'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12629966388

Commit:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12630544171

Commit:
Fix message in tour

robot-ci-heartex commented:
Follow Merge downstream workflow has been failed.
> [Workflow run](https://github.com/HumanSignal/label-studio-enterprise/actions/runs/12641953947)

AndrejOros commented:
/git merge 
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12691538726)
> Unresolvable merge conflict detected in files:
label_studio/core/utils/contextlog.py
Please resolve it using the git command line

niklub committed:
Merge branch 'develop' into fb-dia-1748

Commit:
Poetry lock

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12749097658

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12749326039

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12749663538

robot-ci-heartex committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12750005876
